test/extendedkey_test.exs

```diff
@@ -0,0 +1,665 @@
+defmodule Bitcoinex.Secp256k1.ExtendedKeyTest do
+  use ExUnit.Case
+  doctest Bitcoinex.ExtendedKey
+
+  alias Bitcoinex.ExtendedKey
+
+  @min_hardened_child_num 0x80000000
+  # 2^32 - 1
+  @max_hardened_child_num 0x100000000
+
+  @invalid_xkeys [
+    # changed prefix
+    "zpub661MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjqJoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1EGMcet8",
+    # invalid char
+    "xpubi61MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjqJoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1EGMcet8",
+    # invalid prefix
+    "apub661MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjqJoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1EGMcet8",
+    # invalid len
+    "xpub661MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjqJoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1EGMcet",
+    # invalid len
+    "xpub661MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjqJoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1E",
+    # invalid len (with valid checksum)
+    "Deb7pNXSbX7qSvc2eMjkNYTrggh4pBgYa2QMFjEjj6hUy1iHp71d1gf3ue7Ni7X8Pkhcy13aakoEYSMS1DFiMaXB4qYhN33dh3oqD9n1YWSxVk"
+  ]
+
+  @bip32_test_case_1 %{
+    # test vectors from bip32: https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#test-vectors
+    seed: "000102030405060708090a0b0c0d0e0f",
```

Comment:
In the test cases, we have seed, but we do not test derivation from the seed enough.

The test coverage for parsing looks good. Would be nice to test derivation from seed for all test cases.

Reply:
I added testing for the seed -> master privkey. What did you mean "for all test cases"?

Reply:
Using the bip32 test cases as a crutch, assert that the expected child matches the derived child.

For example, make sure our derivation works from seed to the deepest child provided in the test cases.
seed -> master privkey -> xpub_m_0h_1_2h -> xpub_m_0h_1_2h_2_1000000000

Reply:
yes, makes sense. ill add now